src/ui-test/tests/command.palette.test.ts

```diff
@@ -0,0 +1,101 @@
+    this.timeout(240000);
+
+    //const CAMEL_RUN_DEBUG_ACTION_COMMAND = 'Run Camel Application with JBang and Debug';
+    const CAMEL_RUN_ACTION_COMMAND = 'Run Camel Application with JBang';
+    const CAMEL_ROUTE_YAML_WITH_SPACE = 'demo route.camel.yaml';
+
+    const TEST_ARRAY_RUN = [
+        "Routes startup",
+        "Hello Camel from yaml"];
+
+    let activityBar: ActivityBar;
+    let editorView: EditorView;
+    let driver: WebDriver;
+
+    before('Before setup', async function () {
+        driver = VSBrowser.instance.driver;
+    });
+
+    after('After cleanup', async function () {
+        await new EditorView().closeAllEditors();
+    });
+
+    beforeEach('Before each test', async function () {
+        activityBar = new ActivityBar();
+        await (await activityBar.getViewControl('Explorer')).openView();
+        await VSBrowser.instance.openResources(path.resolve('src', 'ui-test', 'resources'));
+        await VSBrowser.instance.waitForWorkbench();
+
+        const section = await new SideBarView().getContent().getSection('resources');
+        await section?.openItem(CAMEL_ROUTE_YAML_WITH_SPACE);
+
+        editorView = new EditorView();
+        await section.getDriver().wait(async function () {
+            return (await editorView.getOpenEditorTitles()).find(title => title === CAMEL_ROUTE_YAML_WITH_SPACE);
+        }, 5000);
+    });
+
+    afterEach('After each test', async function () {
+        const bottomBarPanel = new BottomBarPanel();
+        const terminal = await bottomBarPanel.openTerminalView();
+        await terminal.killTerminal();
+        await bottomBarPanel.toggle(false);
+    });
+
+    it(`Execute command "${CAMEL_RUN_ACTION_COMMAND}" in command palette`, async function () {
+        await useCommandInPalette(CAMEL_RUN_ACTION_COMMAND);
+        await terminalViewHasText(driver, TEST_ARRAY_RUN);
+    });
+
+    /**
+     * Executes a command in the command prompt of the workbench.
+     * @param command The command to execute.
+     * @returns A Promise that resolves when the command is executed.
+     */
+    async function useCommandInPalette(command: string): Promise<void> {
+        const workbench = new Workbench();
+        await workbench.openCommandPrompt();
+        const input = await InputBox.create();
+        await input.setText(`>${command}`);
+        await input.confirm();
+    }
+
+    /**
+     * Checks if the terminal view has the specified texts in the given textArray.
+     * @param driver The WebDriver instance to use.
+     * @param textArray An array of strings representing the texts to search for in the terminal view.
+     * @param interval (Optional) The interval in milliseconds to wait between checks. Default is 1000ms.
+     * @returns A Promise that resolves to a boolean indicating whether the terminal view has the texts or not.
+     */
+    async function terminalViewHasText(driver: WebDriver, textArray: string[], interval = 1000): Promise<void> {
```

Comment:
that would be nice to extract into utils.ts to allow reuse between tests which checks terminal view output